builtin/game/item_entity.lua

```diff
@@ -189,16 +190,16 @@ core.register_entity(":__builtin:item", {
 		-- Push item out when stuck inside solid node
 		if is_stuck then
 			local shootdir
-			local cx = pos.x % 1
-			local cz = pos.z % 1
+			local cx = (pos.x % 1) - 0.5
+			local cz = (pos.z % 1) - 0.5
 			local order = {}
 
 			-- First prepare the order in which the 4 sides are to be checked.
 			-- 1st: closest
 			-- 2nd: other direction
 			-- 3rd and 4th: other axis
 			local cxcz = function(o, cw, one, zero)
```

Comment:
Our code-style guidelines advise against using this form of function definition, unless defining a function within a table:

> Write function definitions of the form `function foo()` instead of the lambda form `foo = function()`, except when inserting functions in tables inline, where only the second form will work.